Snowflake.Data.Tests/IntegrationTests/StructuredTypesIT.cs

```diff
@@ -0,0 +1,1471 @@
+            }
+        }
+
+        [Test]
+        public void TestSelectMapOfLists()
+        {
+            // arrange
+            using (var connection = new SnowflakeDbConnection(ConnectionString))
+            {
+                connection.Open();
+                using (var command = connection.CreateCommand())
+                {
+                    EnableStructuredTypes(connection);
+                    var mapWithArrayValueSFString = "OBJECT_CONSTRUCT('a', ARRAY_CONSTRUCT('b', 'c'))::MAP(VARCHAR, ARRAY(TEXT))";
+                    command.CommandText = $"SELECT {mapWithArrayValueSFString}";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+
+                    // act
+                    var map = reader.GetMap<string, List<string>>(0);
+
+                    // assert
+                    Assert.AreEqual(1, map.Count);
+                    CollectionAssert.AreEqual(new string[] {"a"}, map.Keys);
+                    CollectionAssert.AreEqual(new string[] {"b", "c"}, map["a"]);
+                }
+            }
+        }
+
+        [Test]
+        public void TestSelectMapOfMaps()
+        {
+            // arrange
+            using (var connection = new SnowflakeDbConnection(ConnectionString))
+            {
+                connection.Open();
+                using (var command = connection.CreateCommand())
+                {
+                    EnableStructuredTypes(connection);
+                    var mapAsSFString = "OBJECT_CONSTRUCT('a', OBJECT_CONSTRUCT('b', 'c'))::MAP(TEXT, MAP(TEXT, TEXT))";
+                    command.CommandText = $"SELECT {mapAsSFString}";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+
+                    // act
+                    var map = reader.GetMap<string, Dictionary<string, string>>(0);
+
+                    // assert
+                    Assert.AreEqual(1, map.Count);
+                    var nestedMap = map["a"];
+                    Assert.AreEqual(1, nestedMap.Count);
+                    Assert.AreEqual("c", nestedMap["b"]);
+                }
+            }
+        }
+
+        [Test]
+        public void TestSelectAllUnstructuredTypesObject()
+        {
+            // arrange
+            using (var connection = new SnowflakeDbConnection(ConnectionString))
+            {
+                connection.Open();
+                using (var command = connection.CreateCommand())
+                {
+                    EnableStructuredTypes(connection);
+                    var timeZone = GetTimeZone(connection);
+                    var expectedOffset = timeZone.GetUtcOffset(DateTime.Parse("2024-07-11 14:20:05"));
+                    var expectedOffsetString = ToOffsetString(expectedOffset);
+                    var allTypesObjectAsSFString = @"OBJECT_CONSTRUCT(
+                        'StringValue', 'abc',
```

Comment:
What about BINARY?

Reply:
I made reading BINARY as byte[]. I can add it as char[] and string as well...